source/MaterialXGenShader/Util.h

```diff
@@ -87,6 +88,17 @@ ValueElementPtr findNodeDefChild(const string& path, DocumentPtr doc, const stri
 void mapNodeDefToIdentiers(const std::vector<NodePtr>& nodes,
                            std::unordered_map<string, unsigned int>& ids);
 
+/// Find lights to use based on an input document
+/// @param doc Document to scan for lights
+/// @param lights List of lights found in document
+void findLights(DocumentPtr doc, std::vector<NodePtr>& lights);
+
+/// Register light node definitions and light count with a given generation context
+/// @param doc Document containing light nodes and definitions
+/// @param lights Lights to register
+/// @param context Context to update
+void registerLights(DocumentPtr doc, const std::vector<NodePtr>& lights, GenContext& context);
```

Comment:
I think maybe these functions belong in the MaterialXRender project? Handling of light source objects is a client side responsibility and will differ much between different targets, so I'm not sure the utilities here will be generally used to belong in the core MaterialXGenShader project.

Reply:
This is the split we chatted about. This part will allow the code generation to include the per light-type code (and set the maximum light preprocessor define. The client cannot insert this code or change the preprocessor value later on, only perform the binding to the light data structures to use the appropriate code. The bind is part of the light handler. This was why I was wondering if we want a "LightBinder" class and make the remapping / registration methods virtual? This code here  would be an "example" binder.